test/integration/kueuectl/list_test.go

```diff
@@ -0,0 +1,199 @@
+	})
+
+	ginkgo.When("List LocalQueue", func() {
+		ginkgo.It("Should print error message with namespace without resources", func() {
+			streams, _, output, errOutput := genericiooptions.NewTestIOStreams()
+			configFlags := CreateConfigFlagsWithRestConfig(cfg, streams)
+			kueuectl := app.NewKueuectlCmd(app.KueuectlOptions{ConfigFlags: configFlags, IOStreams: streams})
+
+			kueuectl.SetArgs([]string{"list", "localqueue", "--namespace", ns.Name})
+			err := kueuectl.Execute()
+
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%s", err)
+			gomega.Expect(output.String()).Should(gomega.BeEmpty())
+			gomega.Expect(errOutput.String()).Should(gomega.Equal(fmt.Sprintf("No resources found in %s namespace.\n", ns.Name)))
+		})
+
+		ginkgo.It("Should print local queue list with all namespaces", func() {
+			streams, _, output, errOutput := genericiooptions.NewTestIOStreams()
+			configFlags := CreateConfigFlagsWithRestConfig(cfg, streams)
+			kueuectl := app.NewKueuectlCmd(app.KueuectlOptions{ConfigFlags: configFlags, IOStreams: streams})
+
+			kueuectl.SetArgs([]string{"list", "localqueue", "--all-namespaces"})
+			err := kueuectl.Execute()
+
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%s: %s", err, output)
+			gomega.Expect(errOutput.String()).Should(gomega.BeEmpty())
+			gomega.Expect(output.String()).Should(gomega.ContainSubstring(ns1.Name))
+			gomega.Expect(output.String()).Should(gomega.ContainSubstring(ns2.Name))
+			gomega.Expect(output.String()).Should(gomega.ContainSubstring(cq1.Name))
+			gomega.Expect(output.String()).Should(gomega.ContainSubstring(cq2.Name))
+			gomega.Expect(output.String()).Should(gomega.ContainSubstring(lq1.Name))
+			gomega.Expect(output.String()).Should(gomega.ContainSubstring(lq2.Name))
+		})
+
+		ginkgo.It("Should print local queue list with filter by cluster queue and namespace", func() {
+			streams, _, output, errOutput := genericiooptions.NewTestIOStreams()
+			configFlags := CreateConfigFlagsWithRestConfig(cfg, streams)
+			kueuectl := app.NewKueuectlCmd(app.KueuectlOptions{ConfigFlags: configFlags, IOStreams: streams})
+
+			kueuectl.SetArgs([]string{"list", "localqueue", "--clusterqueue", cq1.Name, "--namespace", ns1.Name})
+			err := kueuectl.Execute()
+
+			gomega.Expect(err).NotTo(gomega.HaveOccurred(), "%s: %s", err, output)
+			gomega.Expect(errOutput.String()).Should(gomega.BeEmpty())
+			gomega.Expect(output.String()).ShouldNot(gomega.ContainSubstring(ns1.Name))
+			gomega.Expect(output.String()).ShouldNot(gomega.ContainSubstring(ns2.Name))
+			gomega.Expect(output.String()).Should(gomega.ContainSubstring(cq1.Name))
+			gomega.Expect(output.String()).ShouldNot(gomega.ContainSubstring(cq2.Name))
+			gomega.Expect(output.String()).Should(gomega.ContainSubstring(lq1.Name))
+			gomega.Expect(output.String()).ShouldNot(gomega.ContainSubstring(lq2.Name))
+		})
+
+		ginkgo.It("Should print local queue list with all namespaces and filter by cluster queue", func() {
```

Comment:
Only cover in integration the common use case or cases where there might be interaction of multiple controllers.

For simple things like filtering, we can rely on unit tests.

The rationale is that integration tests are expensive.

Reply:
Done